llvm/lib/Target/AIE/AIELegalizerInfo.cpp

```diff
@@ -388,6 +388,17 @@ AIELegalizerInfo::AIELegalizerInfo(const AIEBaseSubtarget &ST) {
       })
       .widenScalarToNextPow2(0)
       .lowerIfMemSizeNotPow2()
+      .bitcastIf(
+          [=](const LegalityQuery &Query) {
+            const LLT &Ty = Query.Types[0];
+            return Ty.isVector() && Ty.getSizeInBits() == 64;
+          },
+          [=](const LegalityQuery &Query) {
+            const LLT Ty = Query.Types[0];
+            unsigned Size = Ty.getSizeInBits();
+            assert(Size % 64 == 0);
```

Comment:
Please add a string with assert, helps in debugging.

Reply:
Agreed. Also, just use `Size` in `LLT:scalar(Size)` below.
That also prevents an unused variable warning in non-assert builds.

Reply:
TBH, I think assert generates a perfectly readable string for this condition. Does it really need to be cluttered with "type size is not a multiple of 64 bits", effectively saying the same?

Reply:
I agree with @martien-de-jong. I will make the "Size" change though..